storage/disk.js

```diff
@@ -1,66 +1,85 @@
-var fs = require('fs')
-var os = require('os')
-var path = require('path')
-var crypto = require('crypto')
-var mkdirp = require('mkdirp')
+const fs = require('fs')
+const os = require('os')
+const path = require('path')
+const crypto = require('crypto')
+const mkdirp = require('mkdirp')
+const { promisify } = require('util')
 
-function getFilename (req, file, cb) {
-  crypto.randomBytes(16, function (err, raw) {
-    cb(err, err ? undefined : raw.toString('hex'))
-  })
+const randomBytes = promisify(crypto.randomBytes)
+const unlink = promisify(fs.unlink)
+
+function getFilename (req, file) {
+  return randomBytes(16).then((raw) => raw.toString('hex'))
 }
 
-function getDestination (req, file, cb) {
-  cb(null, os.tmpdir())
+function getDestination (req, file) {
+  return Promise.resolve(os.tmpdir())
 }
 
-function DiskStorage (opts) {
-  this.getFilename = (opts.filename || getFilename)
+class DiskStorage {
+  constructor (opts = {}) {
+    this.getFilename = opts.filename || getFilename
+    this.maxSize = opts.maxSize || Infinity 
```

Comment:
There seems to be a trailing space at the end and standard is unable to fix it automatically. Please remove it.